services/thermal_mgr/thermal_mgr.c

```diff
@@ -41,20 +41,46 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 }
 
+/**
+ * @brief Sends event to thermal manager queue
+ * 
+ * @param event - Pointer to thermal manager event to send
+ * @return ERR_CODE_SUCCESS if the event is successfully sent
+ */
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  xQueueSend(thermalMgrQueueHandle, event, 0);
   return ERR_CODE_SUCCESS;
 }
 
+/**
+ * @brief OS interrupt handler for LM75B temperature sensor
+ * 
+ */
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  float temperature;
+  readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temperature);
+  if(temperature > 75.0){ // only below T hys is safe
+    overTemperatureDetected();
+  }
+  else safeOperatingConditions();
 }
 
+/**
+ * @brief Thermal management task for temperature events
+ * 
+ * @param pvParameters - Pointer to parameters passed to task
+ */
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t data = *(thermal_mgr_event_t *) pvParameters;
```

Comment:
In this case, pvParameters isnt of type thermal_mgr_event_t but of the config which isn't the same. So I would just change the line to initialize an empty event instead